applications/user-api/src/main/java/no/fdk/userapi/LocalUserService.java

```diff
@@ -0,0 +1,67 @@
+package no.fdk.userapi;
+
+import com.fasterxml.jackson.databind.ObjectMapper;
+import org.springframework.stereotype.Component;
+
+import java.io.BufferedReader;
+import java.io.InputStreamReader;
+import java.net.URL;
+import java.net.URLConnection;
+import java.util.Collections;
+import java.util.List;
+import java.util.stream.Collectors;
+
+@Component
+public class LocalUserService {
+    private String URL_BASE_STRING = "https://publishers-api.ut1.fellesdatakatalog.brreg.no/domains/";
+    private String URL_END = "/organizations";
+
+    User getUser(String id) {
+        String domain = id.substring(id.indexOf("@") + 1);
+        List<String> organisations = getOrganisationsAssociatedWithDomain(domain);
+
+        return new LocalUserAdapter(id, organisations);
+    }
+
+    private List<String> getOrganisationsAssociatedWithDomain(String domain) {
+        try {
+            URLConnection connection = new URL(URL_BASE_STRING + domain + URL_END).openConnection();
+            connection.setRequestProperty("Accept", "application/json");
+            try (BufferedReader reader = new BufferedReader(new InputStreamReader(connection.getInputStream()))) {
+                String jsonBody = reader.lines().collect(Collectors.joining(System.lineSeparator()));
+                ObjectMapper mapper = new ObjectMapper();
+                return mapper.readValue(jsonBody, List.class);
+            }
+        } catch (Exception e) {
+            return Collections.emptyList();
+        }
+    }
+
+    private class LocalUserAdapter implements User {
+        private String id;
+        private List<String> organisations;
+
+        LocalUserAdapter(String id, List<String> organisations) {
+            this.id = id;
+            this.organisations = organisations;
+        }
+
+        public String getId() {
+            return this.id;
+        }
+
+        public String getFirstName() {
+            return "";
+        }
+
+        public String getLastName() {
+            return "";
+        }
+
+        public String getAuthorities() {
+            List<String> privilegesList = organisations.stream()
+                .map(o -> "publisher:" + o + ":read").collect(Collectors.toList());
```

Comment:
den kolonlogik er komplisert og duplisert i AltinnUserService., kanskje ha en ResourceRole class constructor som tar tre parametrer?, eler PublisherRole som tar to? eller PublisherReadRole som tar en? eller kombinere alle konstruktorene som statisk metod av Authority klass.

Reply:
Min plan var å flytte alt inn i en egen Authorities/Authority-klasse